trestle/core/catalog/catalog_merger.py

```diff
@@ -0,0 +1,145 @@
+
+            Reading the markdown tells you groups and controls in them - and groups in groups.
+            Controls cannot change groups.  If the control was in the original json, its parts are replaced,
+            including its parameters.  Only values may be specified.  If no value specified, the value is unset in json.
+        """
+        cat_interface = CatalogInterface(catalog)
+        for src in cat_interface.get_all_controls_from_dict():
+            group_id, _, _ = cat_interface.get_group_info_by_control(src.id)
+            dest = self._catalog_interface.get_control(src.id)
+            if dest:
+                dest_group, _, _ = self._catalog_interface.get_group_info_by_control(dest.id)
+                if dest_group != group_id:
+                    raise TrestleError(f'Markdown for control {src.id} has different group id.')
+                CatalogMerger.merge_controls(dest, src, replace_params)
+                self._catalog_interface.replace_control(dest)
+            else:
+                # this is a new control that isn't already in the merge destination
+                # need to add the control knowing its group must already exist
+                # get group info from an arbitrary control already present in group
+                _, control_handle = self._catalog_interface._find_control_in_group(group_id)
+                new_control_handle = copy.deepcopy(control_handle)
+                new_control_handle.control = src
+                # add the control and its handle to the param_dict
+                self._catalog_interface._control_dict[src.id] = new_control_handle
+
+        # now need to cull any controls that are not in the src catalog
+        handled_ids = set(cat_interface._control_dict.keys())
+        orig_ids = set(self._catalog_interface._control_dict.keys())
+        extra_ids = orig_ids.difference(handled_ids)
+        for extra_id in extra_ids:
+            self._catalog_interface._control_dict.pop(extra_id)
+
+        self._catalog_interface.update_catalog_controls()
+
+    def _merge_header_and_comp_dict(
+        self, control: cat.Control, control_file_path: pathlib.Path, context: ControlContext
+    ) -> None:
+        """
+        Merge the header and the comp_dict.
+
+        Notes:
+            now have all rules in context.rules_dict and all rules_params in context.rules_params_dict
+            all set-params per component for each control are in the cat interface
+            all comp-infos by control and part are in the cat interface
+
+            can now write out catalog and pull from the markdown:
+            header for param values to set during assem
+            prose and status for This System
+            status for all parts that still have rules
+        """
+        memory_header, memory_comp_dict = self._catalog_interface._get_control_memory_info(control.id, context)
+        # FIXME confirm this merge behavior
```

Comment:
This needs to be confirmed

Reply:
This is a reminder to check the current rules for what can and cannot be edited in markdown and needs hands on testing.  So I will leave the fixme for now.